### PR TITLE
[XPU] Add gpt-oss model support for Intel GPU

### PR DESCRIPTION
**Pull request by jikunshang**
## Purpose
this PR introduce a new `IpexFp4MoeMethod` for xpu, which support Wfp4A16 moe_gemm, implemented in ipex library.  With that we can run openai/gpt-oss-20b, openai/gpt-oss-120b on Intel B60 GPU.

gpt-oss-20b accuracy:
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     0|exact_match|?  |0.9333|±  |0.0069|
|     |       |strict-match    |     0|exact_match|?  |0.8264|±  |0.0104|

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/mxfp4.py [hunk 1]
```diff
@@ -1146,3 +1155,83 @@ def apply(
...
+        x_pad = torch.nn.functional.pad(x, (0, hidden_size_pad - x.size(-1)))
+        hidden_states = layer.ipex_fusion(
+            x_pad,
+            use_grouped_topk,
+            top_k,
+            router_logits,
+            renormalize,
+            topk_group,
+            num_expert_group,
+            activation="swiglu_oai",
```
[the remark this answers is not in the document]
Comment (jikunshang) [reply vllm/model_executor/layers/quantization/mxfp4.py:1235]: a little difference in naming, vllm use `swigluoai`, ipex interface accept `swiglu_oai`, I prefer to keep this.

File: vllm/model_executor/layers/quantization/mxfp4.py [hunk 2]
```diff
@@ -188,7 +191,10 @@ def get_quant_method(
                 return UnquantizedLinearMethod()
             raise NotImplementedError("Mxfp4 linear layer is not implemented")
         elif isinstance(layer, FusedMoE):
-            return Mxfp4MoEMethod(layer.moe_config)
+            if current_platform.is_xpu():
+                return IpexFp4MoeMethod(layer.moe_config)
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/mxfp4.py]: Nit: please use mxfp4 explicitly i.e. `IpexMxfp4MoEMethod`

File: vllm/model_executor/layers/quantization/mxfp4.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgoin) [reply vllm/model_executor/layers/quantization/mxfp4.py:1235]: @jikunshang can you add an assert that activation is swiglu_oai then?

**Review by mgoin — COMMENTED** [review]
Seems reasonable, thanks! Just some nits

File: vllm/model_executor/layers/quantization/mxfp4.py (hunk 2 above)
Comment (jikunshang) [reply vllm/model_executor/layers/quantization/mxfp4.py]: updated. thanks!

File: vllm/model_executor/layers/quantization/mxfp4.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jikunshang) [reply vllm/model_executor/layers/quantization/mxfp4.py:1235]: sure, please take a look again:)

File: vllm/model_executor/layers/quantization/mxfp4.py [hunk 3]
```diff
@@ -1146,3 +1155,86 @@ def apply(
...
         else:
             raise ValueError(f"Unsupported backend: {self.mxfp4_backend}")
+
+
+class IpexMxfp4MoEMethod(Mxfp4MoEMethod):
+    def __init__(self, moe_config: FusedMoEConfig):
+        super().__init__(moe_config)
+        self.moe_config = moe_config
+        self.alpha = 1.702
+        self.limit = 7.0
```
Comment (Isotr0py) [inline vllm/model_executor/layers/quantization/mxfp4.py]:
```suggestion
```
Seems these fields from activation unused?

Comment (jikunshang) [reply vllm/model_executor/layers/quantization/mxfp4.py]: nice catch, removed.

**Review by Isotr0py — APPROVED** [review]
